### PR TITLE
LCNV-30932: Vehicle last known stats API

### PR DESCRIPTION
**Pull request by gaurav4ever**
**What**
- Vehicle last known stats API

JIRA: https://babatrucks.atlassian.net/browse/LCNV-30932
Ref sheet: https://docs.google.com/spreadsheets/d/1q3PIe9TAj45ibQQuL-Ro2rdOQmB5q_v6hxh5V7-0KUI/edit#gid=559317985

### CODE REVIEW

File: app/services/sensor/sensor_service.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      sensor_count = {}
+      sensor_types.each do |type|
+        return [false, "Sensor type #{type} not supported"] unless sensor_type_mapping[type]
+
+        get_gps_sensors(sensor) if type == "gps"
+
+        sensor << type
+
+        sensor_count[sensor_type_mapping[type]] = "sensor"
+        return [false,"Only 3 sensors supported at a time"] if sensor_count.size > 3
```
Comment (gaurav4ever) [inline app/services/sensor/sensor_service.rb:24]: code not formatted

File: app/services/sensor/sensor_service.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    end
+
+    def get_gps_sensors(sensor)
+      get_default_sensor.each do |default_sensor|
+        sensor << default_sensor
+      end
+    end
+
+    def get_default_sensor
+      ["speed", "ignition", "orientation", "current_location_coordinates","gps"]
```
Comment (gaurav4ever) [inline app/services/sensor/sensor_service.rb:51]: code not formatted

File: config/sensor2.yml
Comment (gaurav4ever) [inline config/sensor2.yml:1]: Remove this file. No longer needed

File: app/services/vehicle/telematics/stats_service.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+      private def format_response(sensor_response)
+        sensor_response.map do |vehicle|
+          {
+            vehicle_number: vehicle["vehicle_number"],
+            vehicle_id: vehicle["vehicle_id"],
+          }.merge(
+
+            sensors.each_with_object({}) do |sensor, extracted|
+              if ["speed", "ignition", "orientation", "current_location_coordinates","gps"].include?(sensor)
```
Comment (gaurav4ever) [inline app/services/vehicle/telematics/stats_service.rb:53]: Make these constants.

File: config/sensors.yml [hunk 4]
```diff
@@ -0,0 +1,429 @@
+supported_sensors:
```
Comment (gaurav4ever) [inline config/sensors.yml:1]: Are you using this field ?

Comment (nishantbansal1910) [reply config/sensors.yml:1]: removed

**Pull request closed without merging**